### DISCUSSION TITLE
Leverage recorder to display all recent states in Automation state triggers

### DISCUSSION THREAD

**Discussion by SeanPM5** [home-assistant/frontend#11224] [Other feature requests]
## The request

Here's what the typical triggers looks like in Automation Editor:

![triggers](https://i.imgur.com/xp2t2wD.png)

You select your entity, but you get zero help after that. You either have to be familiar with the various states that entity can have, or you need to cross-reference with Developer Tools > States.

An idea I had was to leverage `recorder:` to suggest states here. If you were to open the more info dialog for the above `sensor.battery_state` entity, you would see the various states it's been in for the past day. So Home Assistant has this information already.

![states](https://i.imgur.com/2dvkfpk.png)

So now here is a quick mockup of the idea.

![recent-states](https://i.imgur.com/GG5XVdA.jpg)

Next to the "From" and "To" text fields would be a dropdown showing the recent states it's been in (from the history that `recorder:` has available). In this case the dropdown would have Charging, Not Charging, Full. If you were to select a state it'd fill the text field with that state.

Design wise this could probably be improved, maybe implement it right into the text field itself, an autosuggest type thing, or display the states as chips underneath or something. This was just a quick thing to get the idea across.

## The alternatives

Alternative would be to display a "more info" icon / button alongside the entity. So to the left or the right of the entity dropdown would be a little "i" icon. This way you would easily be able to find the recent states without heading to Dev Tools > States panel.

**Comment by iantrich** [comment]
Cool idea! integrated into the existing textfield with a "Not Listed" always available as well that would just prompt you to type seems best IMO

**Comment by matthiasdebaat** [comment]
Like the idea! Maybe merge those two fields into one, so there's a list of option you could chose?
